include/rsl/algorithm.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <algorithm>
+#include <vector>
+
+namespace rsl {
+
+/**
+ * @brief Does the collection contain the value
+ * 
+ * @param collection   The input collection
+ * @param value        The value to test for in the collection
+ * 
+ * @return true if the value is found in the collection
+ */
+template<typename T>
+bool does_contain(std::vector<T> const& collection, T const& value) {
+  return std::find(collection.cbegin(), collection.cend(), value) != collection.cend();
+}
```

Comment:
Do these both of these algorithms need to require the use of a vector? Why not an arbitrary random access container?

Reply:
Because the vector is the base data structure of C++ for as much as the STL seems to think it is the iterator.

I didn't want to do this with ranges because I don't understand them well enough to take a parameter that is a view. Maybe I should look into that.

The other reason these take vectors is that the rclcpp::Parameter types that have homogeneous collections in them are vectors, and these are needed for writing certain types of validation functions of them.